### PR TITLE
block s2i from doing ONBUILD operations

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton @mfojtik ptal  (@mfojtik need you to review the env var change i had to make when bumping s2i)

**Comment by smarterclayton** [comment]
Looks ok to me but will defer to Michal on sign off

**Comment by bparees** [comment]
[test]
[testonlyextended][extended:core(s2i)]

**Comment by bparees** [comment]
another bucket to run: core(source-to-image).

File: pkg/build/builder/sti.go [hunk 1]
```diff
@@ -343,11 +344,11 @@ func (d *downloader) Download(config *s2iapi.Config) (*s2iapi.SourceInfo, error)
...
 //    instead of deferring what to do with repeated environment variables to the
 //    Docker runtime.
-func buildEnvVars(build *api.Build) map[string]string {
+func buildEnvVars(build *api.Build) s2iapi.EnvironmentList {
 	bi := buildInfo(build)
-	envVars := make(map[string]string, len(bi))
+	envVars := s2iapi.EnvironmentList{}
 	for _, item := range bi {
-		envVars[item.Key] = item.Value
+		envVars = append(envVars, s2iapi.EnvironmentSpec{Name: item.Key, Value: item.Value})
```
Comment (mfojtik) [inline pkg/build/builder/sti.go]: you can also use `err = envVars.Set("KEY=VALUE")` to make sure the checks for the format/etc apply, but append will work as well.

**Comment by bparees** [comment]
@mfojtik updated per your suggestion, ptal

**Comment by bparees** [comment]
manually tested env var behavior. [merge]

**Comment by mfojtik** [comment]
LGTM

**Comment by bparees** [comment]
[test]
[merge]

**Comment by bparees** [comment]
@smarterclayton @mfojtik @jwforres i ended up doing the bump here because i was going to have to update origin code (as this PR does) to make it compatible w/ the bump i was having to do to revert the change(sigh).

**Comment by bparees** [comment]
[testonlyextended][extended:core(s2i build with environment file in sources)]

**Pull request merged**